camel/agents/chat_agent.py

```diff
@@ -348,6 +352,61 @@ def step(
 
         return ChatAgentResponse(output_messages, self.terminated, info)
 
+    def step_completion(
```

Comment:
Why do we need `step_completion` method if we already have `step`?
THe contents seem to be a simplified duplicate of step. How can we make it DRY?

Reply:
# used for output structure string.
I would like to fill in the blank in the prompt using `OpenAI Completion API` [http:](https://platform.openai.com/docs/api-reference/completions). For example:
## prompt
```
Domain expert 1: <|blank|>
Associated competencies, professional characteristics, duties and workflows: <|blank|>. End
```
And then the completion API will return the response, for example:
```
Domain expert 1: Data Scientist
Associated competencies, professional characteristics, duties and workflows: Mastery in statistical analysis. Proficiency in programming languages such as Python and R. End
```
What is more, you will see the API of `Completion` and `ChatCompletion` is different, so in my personal opinion, it is not a duplication.

Reply:
Now I have removed the model (using `TURBO` rather than `DAVINCI`) according to your comment.